### PR TITLE
CTO-141 project rights

### PR DESCRIPTION
**Pull request by zack-rma**
Taking over PR #195

### CODE REVIEW

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockController.java [hunk 1]
```diff
@@ -1,4 +1,133 @@
...
+import java.util.List;
+import mil.army.usace.hec.cwms.http.client.ApiConnectionInfo;
+import mil.army.usace.hec.cwms.http.client.HttpRequestBuilderImpl;
+import mil.army.usace.hec.cwms.http.client.HttpRequestResponse;
+import mil.army.usace.hec.cwms.http.client.request.HttpRequestExecutor;
+import mil.army.usace.hec.cwms.radar.client.model.ProjectLock;
+import mil.army.usace.hec.cwms.radar.client.model.ProjectLockId;
+import mil.army.usace.hec.cwms.radar.client.model.RadarObjectMapper;
+
 public class ProjectLockController {
```
Comment (adamkorynta) [inline cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockController.java:15]: mark final

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockInput.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            return projectId;
+        }
+    }
+
+    public static final class LockRequest extends EndpointInput {
+        public static final String REVOKE_EXISTING = "revoke-existing";
+
+        private final ProjectLock lock;
+        private boolean revokeExisting = false;
+        private int revokeTimeout = 10;
```
Comment (adamkorynta) [inline cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockInput.java:102]: Variable should carry the units

**Review by adamkorynta — CHANGES_REQUESTED** [review]

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockController.java (hunk 1 above)
Comment (zack-rma) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockController.java]: Marked final

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockInput.java (hunk 2 above)
Comment (zack-rma) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/ProjectLockInput.java]: Added "seconds" to variable name

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**